New Functions, Fixes and Formatting

Robert-Zacchigna opened the pull request:
Hello @jessecooper, I'm back again with some fixes, formatting and two new functions.

Please review at your leisure, I'll leave the version bump up to you.

**ALSO, I think you need to manually run whatever process you have to update the docs site, as it is still showing an older version (preferably after these changes are merged if you approve).**

I tried to see if it was something i could do but i don't think it is (please let me know though).

Thank you.

## Major Changes

* **New Functions**
  * `get_portfolio_lot`
    * Retrieves account portfolio position lot based on provided symbol  (using `get_account_portfolio`)
  * `list_order_details`
    * Lists order details of a specific account ID Key and order ID (retrieved from `list_orders`)


* **Function Updates and Fixes**
  * <ins>Updates</ins>
    * `list_orders`
      * kwargs expanded and new inputs linted
      * Description updated with expanded inputs
  * <ins>Fixes</ins>
    * `list_alerts`
      * List alert limit (`count`) properly enforced
    * `get_quote`
      * Get quote symbol limit (`count`) properly enforced

## Minor Changes

* `Pipenv` removed and replace with `Poetry`
  * Pre-commit-config updates
    * General file formatting from new pre-commit hooks
* Unit-tests updated
* README updated
  * General formatting
  * Include all completed endpoints/functions
  * Updated development setup

jessecooper commented on Makefile:
```diff
@@ -2,11 +2,11 @@ init:
-lint:
-	flake8 pyetrade tests
+analysis: # Lint, format, import optimizer, etc.
+	pipenv run pre-commit run --all-files
```
I would like not to use `pipenv` given the choice. I am not a big fan of the project and the same effect could be accomplished by getting into the venv and running `pre-commit run --all-files`. In the future I might change to something like `poetry` because it uses a little more standard  PEP conventions of just using the `pyproject.toml`.

Robert-Zacchigna replied:
oh i see, no problem. I'm not too familiar with `poetry` but i would be willing to give it a try and set it up for the project (if you don't mind that is).

Also to be clear, this is what you are referring to correct?: https://python-poetry.org/

Robert-Zacchigna replied:
I have removed `pipenv` and replaced with `poetry`. Please see my latest comment and commits. Thank you.

Robert-Zacchigna commented:
@jessecooper I have replaced `pipenv` with poetry, which was actually pretty easy thankfully. I have setup the `.toml` file, updated the pre-commit config with poetry hooks, updated and locked the dependencies using poetry and updated the devel section of the readme.

jessecooper commented on Makefile:
```diff
@@ -2,11 +2,13 @@ init:
 	pip install -r requirements_dev.txt
-	pre-commit install
+	pre-commit install --hook-type pre-push --install-hooks -t post-checkout -t post-merge
+	curl -sSL https://install.python-poetry.org | python3 -
```
What is this curl for? You should be able to just `pip install poetry`?

Robert-Zacchigna replied:
huh you know what you're right. I was looking at the docs for installation and the first one option was [`pipx`](https://python-poetry.org/docs/#installing-with-pipx).

The next one was the curl cmd but now i see under the [`advanced`](https://python-poetry.org/docs/#installing-manually) section you can just use pip. I'll make the change.

Robert-Zacchigna replied:
see latest commits

jessecooper commented on pyetrade/accounts.py:
```diff
@@ -215,16 +277,20 @@ def list_transactions(
+            "startDate": start_date.date().strftime("%m%d%Y")
+            if start_date
+            else start_date,
+            "endDate": end_date.date().strftime("%m%d%Y") if end_date else end_date,
```
This statement looks a little funny. I understand `end_date` would be `None` if it didn't hit the `if` statement but I think it should just be `else None` just so it is more explicit and clear.

Robert-Zacchigna replied:
Ya, looking back, i agree, leaving it like that would only confuse someone who isn't familiar with the code. I'll make the change.

Robert-Zacchigna replied:
See latest commits

Robert-Zacchigna commented:
@jessecooper Ok I have updated the poetry installation method to simply use `pip`

I have also made it explicit when a variable is is `None` and is not being used and updated the supported python versions in `setup.py` (removing 3.8 and adding 3.12).

jessecooper reviewed (approved):
LGTM;

jessecooper commented:
@Robert-Zacchigna Thank you for all your work on this. You have done a great amount of work to modernize and improve this package. Thank you!

jessecooper commented:
Also, I will give a look into the docs issue.

jessecooper commented:
@Robert-Zacchigna Thanks for the heads up on the docs issue. I got that all worked out. Latest docs are up here:
https://pyetrade.readthedocs.io/en/latest/
